### PR TITLE
Fix issue9

### LINKED ISSUE #9: debugging theta['dm'] always converging to zero
Running an experiment with `traj_cost.addThetaCost('dm', Quadratic(n=0, d=0, weight=0.1))` is still resulting in `dm` converging to zero. Below is the code to insert `weight=0.1`.
https://github.com/jsalfity/sensitivity_games/blob/2914adb3e5460203a30f1482f32c7e397ec7b718/sensitivity_games/experiments/point_mass_regulation_train.py#L59

Re-evaluating the cost function calculation, I just want to double check a few things are correct:

- Our method to achieve gradient ascent:
https://github.com/jsalfity/sensitivity_games/blob/2914adb3e5460203a30f1482f32c7e397ec7b718/sensitivity_games/experiments/point_mass_regulation_train.py#L64

- Our method to evaluate the trajectory cost, paying close attention to the negative sign for `theta`
https://github.com/jsalfity/sensitivity_games/blob/2914adb3e5460203a30f1482f32c7e397ec7b718/sensitivity_games/trajectory_cost.py#L26-L39

- In the `Quadratic.evaluate()` function, the nominal value, `n`, for `theta['dm']` is zero.
https://github.com/jsalfity/sensitivity_games/blob/2914adb3e5460203a30f1482f32c7e397ec7b718/sensitivity_games/quadratic.py#L19-L27

I'm having trouble stepping through the logic that the combination of these three, along with others, are pushing `dm` to go to 'high disturbance' value. My thought process is ... we want our cost function to be smaller, except for `dm` to be larger, which justifies bullet 1 and 2. I'm can't rationalize why bullet 3 is correct, do we really want a `theta[dm]` nominal=0 ?

We wrote some math out , making sure we are still solving this same problem:
Screen Shot 2021-07-28 at 12 08 16 PM

Another option I was thinking was to add friction or wind resistance, but then of course it wouldn't be a linear problem.

### PR DESCRIPTION
**Pull request by jsalfity**
Main update is updating `point_mass.modify_B()` function:
https://github.com/jsalfity/sensitivity_games/blob/f64b886a29a4a05b503cb2e55d431a49a32bcf14/sensitivity_games/point_mass.py#L83-L95

Another update is creating the option of static or random initial conditions.

https://github.com/jsalfity/sensitivity_games/blob/f64b886a29a4a05b503cb2e55d431a49a32bcf14/sensitivity_games/experiments/point_mass_regulation_train.py#L71-L84

Closes #9

### CODE REVIEW

File: sensitivity_games/linear_feedback_controller.py [hunk 1]
```diff
@@ -22,19 +22,21 @@ def solve_dare_for_k(self, Q, R):
...
         X = torch.tensor(linalg.solve_discrete_are(self.dynamics.A,
-                                                   self.dynamics.B,
+                                                   self.dynamics.B.detach(),
                                                    Q,
                                                    R), dtype=torch.float32)
+
         # compute the LQR gain
-        term1 = torch.tensor(linalg.inv(self.dynamics.B.T@X@self.dynamics.B+R),
-                             dtype=torch.float32)
+        inv = linalg.inv((self.dynamics.B.T@X@self.dynamics.B).detach()+R)
```
Comment (dfridovi) [inline sensitivity_games/linear_feedback_controller.py]: nit: add whitespace around arithmetic operations, i.e. `B.T @ X @ B`

Also, it's faster and more numerically stable to do something like `torch.linalg.solve(B.T @ X @ B, term2)` to get `K`.

**Review by dfridovi — APPROVED** [review]
LGTM! I think we're all set to merge + work on NN controller replacing `K`.

File: sensitivity_games/linear_feedback_controller.py (hunk 1 above)
Comment (jsalfity) [reply sensitivity_games/linear_feedback_controller.py]: great. i'll change a bit. i'm new to working with `flake8 `, which caused me to break apart the calculation.
